test/Staking.test.js

```diff
@@ -0,0 +1,299 @@
+        let balanceBeforeOwner = await instanceToken.balanceOf(owner);
+        expect(balanceBeforeOwner).to.be.bignumber.equal(ether('5600000'));
+        let balanceBeforeContract = await instanceToken.balanceOf(instanceStaking.address);
+        expect(balanceBeforeContract).to.be.bignumber.equal(ether('0'));
+        let start = await time.latest();
+        await instanceToken.approve(instanceStaking.address, ether('500000'));
+        await expectRevert(instanceStaking.setRewards(start.add(new BN(60)), ether('5000000'), ether('10')), "Error : reward above limit - 500000");
+        let balanceAfterOwner = await instanceToken.balanceOf(owner);
+        expect(balanceAfterOwner).to.be.bignumber.equal(ether('5600000'));
+        let balanceAfterContract = await instanceToken.balanceOf(instanceStaking.address);
+        expect(balanceAfterContract).to.be.bignumber.equal(ether('0'));
+      });
+      it("SetRewards false - Error : start time must be greater than current time", async () => {
+        let balanceBeforeOwner = await instanceToken.balanceOf(owner);
+        expect(balanceBeforeOwner).to.be.bignumber.equal(ether('5600000'));
+        let balanceBeforeContract = await instanceToken.balanceOf(instanceStaking.address);
+        expect(balanceBeforeContract).to.be.bignumber.equal(ether('0'));
+        let start = await time.latest();
+        await instanceToken.approve(instanceStaking.address, ether('500000'));
+        await expectRevert(instanceStaking.setRewards(start.add(new BN(0)), ether('500000'), ether('10')), "Error : start time must be greater than current time");
+        let balanceAfterOwner = await instanceToken.balanceOf(owner);
+        expect(balanceAfterOwner).to.be.bignumber.equal(ether('5600000'));
+        let balanceAfterContract = await instanceToken.balanceOf(instanceStaking.address);
+        expect(balanceAfterContract).to.be.bignumber.equal(ether('0'));
+      });
+      it("SetRewards false - Error : one of params equal to 0", async () => {
+        let balanceBeforeOwner = await instanceToken.balanceOf(owner);
+        expect(balanceBeforeOwner).to.be.bignumber.equal(ether('5600000'));
+        let balanceBeforeContract = await instanceToken.balanceOf(instanceStaking.address);
+        expect(balanceBeforeContract).to.be.bignumber.equal(ether('0'));
+        let start = await time.latest();
+        await instanceToken.approve(instanceStaking.address, ether('500000'));
+        await expectRevert(instanceStaking.setRewards(start.add(new BN(60)), ether('500000'), new BN(0)), "Error : one of params equal to 0");
+        let balanceAfterOwner = await instanceToken.balanceOf(owner);
+        expect(balanceAfterOwner).to.be.bignumber.equal(ether('5600000'));
+        let balanceAfterContract = await instanceToken.balanceOf(instanceStaking.address);
+        expect(balanceAfterContract).to.be.bignumber.equal(ether('0'));
+      });
+      it("SetRewards false - Ownable: caller is not the owner", async () => {
```

Comment:
rename to Should fail if caller is not owner, shouldn't setup error message in desceribe